From 20260813-susfs-uname Mon Sep 17 00:00:00 2001
From: Pulsar Bot <pulsar@localhost>
Date: Thu, 13 Aug 2026 13:41:00 -0300
Subject: [PATCH] susfs: adapt spoof_uname hook to Pulsar 4.19 uname

Add the SuSFS uname hook to the existing Pulsar 4.19 uname implementation.

Signed-off-by: Pulsar Bot <pulsar@localhost>
---
 kernel/sys.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/kernel/sys.c b/kernel/sys.c
--- a/kernel/sys.c
+++ b/kernel/sys.c
@@ -1239,6 +1239,9 @@ static int override_release(char __user *release, size_t len)
 	return ret;
 }
 
+#ifdef CONFIG_KSU_SUSFS_SPOOF_UNAME
+extern void susfs_spoof_uname(struct new_utsname *tmp);
+#endif
 SYSCALL_DEFINE1(newuname, struct new_utsname __user *, name)
 {
 	struct new_utsname tmp;
@@ -1248,6 +1251,8 @@ SYSCALL_DEFINE1(newuname, struct new_utsname __user *, name)
 
 	down_read(&uts_sem);
 	memcpy(&tmp, utsname(), sizeof(tmp));
+#ifdef CONFIG_KSU_SUSFS_SPOOF_UNAME
+	susfs_spoof_uname(&tmp);
+#endif
 	if (cur_uid == 0 &&
 		(!strncmp(current->comm, "bpfloader", 9) ||
